src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -471,9 +498,29 @@ private async void RunLeaseRenewalLoopAsync()
                     await connection.OpenAsync(token);
                     this._logger.LogDebugWithThreadId("END OpenLeaseRenewalLoopConnection");
 
+                    bool forceReconnect = false;
                     while (!token.IsCancellationRequested)
                     {
-                        await this.RenewLeasesAsync(connection, token);
+                        bool isConnected = await connection.TryEnsureConnected(forceReconnect, this._logger, "LeaseRenewalLoopConnection", token);
+                        if (!isConnected)
+                        {
+                            // If we couldn't reconnect then wait our delay and try again
+                            await Task.Delay(TimeSpan.FromSeconds(LeaseRenewalIntervalInSeconds), token);
+                            continue;
+                        }
+                        else
+                        {
+                            forceReconnect = false;
+                        }
+                        try
+                        {
+                            await this.RenewLeasesAsync(connection, token);
```

Comment:
Is retrying safe here? 
Do you have checks to ensure it doesn't cause data redundancy if server DOES process query in first attempt even though connection terminates due to network failure?

Reply:
It should be, yes. All queries are done in transactions so any errors will result in a rollback and retry next time around.